lib/data/data_sources/remote_data_source/users_data_source.dart

```diff
@@ -1,23 +1,122 @@
+import 'dart:async';
 import 'package:sendbird_sdk/sendbird_sdk.dart';
+import '../../../Core/constants.dart';
+import 'user_batch_data_entry.dart';
+import 'package:sendbird_sdk/query/user_list/user_list_query.dart' as sendQuery;
+import '../../../domain/entities/chat_user.dart';
+import '../local_data_source/user_type_data_source.dart';
+import 'models/user.dart';
 
 class UsersDataSource {
   final SendbirdSdk sendbird;
-  UsersDataSource({this.sendbird});
+  final UserTypeDataSource localUserTypeDataSource;
 
-  Future<List<User>> getUsers() {
+  UsersDataSource({
+    required this.sendbird,
+    required this.localUserTypeDataSource,
+  });
+
+  Future<List<User>> getUsers() async {
+    await sendbird.connect(Constants.adminUserId,
+        nickname: Constants.adminNickName);
+    final query = sendQuery.ApplicationUserListQuery();
     try {
-      final query = ApplicationUserListQuery();
-      return query.loadNext();
+      var list = await query.loadNext();
+      if (list.length <= 2) await createUsers();
+      return list;
     } catch (e) {
-      throw Exception(e);
+      throw Exception(e.toString());
```

Comment:
not sure how we are handling the exceptions between layers but this seems to be wrong